shared/src/commonMain/kotlin/org/hyperskill/app/providers/domain/interactor/ProvidersInteractor.kt

```diff
@@ -0,0 +1,14 @@
+package org.hyperskill.app.providers.domain.interactor
+
+import org.hyperskill.app.providers.domain.model.Provider
+import org.hyperskill.app.providers.domain.repository.ProvidersRepository
+
+class ProvidersInteractor(
+    private val providersRepository: ProvidersRepository
+) {
+    suspend fun getProviders(providersIds: List<Long>, forceLoadFromRemote: Boolean): Result<List<Provider>> =
+        providersRepository.getProviders(providersIds, forceLoadFromRemote)
+
+    suspend fun getProvider(providerId: Long, forceLoadFromRemote: Boolean): Result<Provider> =
+        providersRepository.getProvider(providerId, forceLoadFromRemote)
+}
```

Comment:
I don't know, do we need a dedicated `ProvidersInteractor`. It contains nothing but repository's calls.

Reply:
@XanderZhu let's leave it for now, depends on how we will implement `TrackSelectionDetailsFeature` and `ProjectSelectionDetailsFeature`:

1. It's needed if we will use it in the `TrackSelectionDetailsActionDispatcher` or `ProjectSelectionDetailsActionDispatcher`.
2. If we will use `TrackSelectionDetailsInteractor` or `ProjectSelectionDetailsInteractor` which under the hood will refer to repositories and load all necessary data, then we don't need `ProvidersInteractor`.

cc @vladkash